README.md

```diff
@@ -260,7 +260,9 @@ A Service Binding describes the connection between a [Provisioned Service](#prov
 
 Restricting service binding to resources within the same namespace is strongly **RECOMMENDED**.  Implementations that choose to support cross-namespace service binding **SHOULD** provide a security model that prevents attacks like privilege escalation and secret enumeration, as well as a deterministic way to declare target namespaces.
 
-A Service Binding resource **MUST** define a `.spec.application` which is an `ObjectReference`-like declaration.  A `ServiceBinding` **MAY** define the application reference by-name or by-[label selector][ls]. A name and selector **MUST NOT** be defined in the same reference.  A Service Binding resource **MUST** define a `.spec.service` which is an `ObjectReference`-like declaration to a Provisioned Service-able resource.  Extensions and implementations **MAY** allow additional kinds of applications and services to be referenced.
+A Service Binding resource **MUST** define a `.spec.application` which is an `ObjectReference`-like declaration.  A `ServiceBinding` **MAY** define the application reference by-name or by-[label selector][ls]. A name and selector **MUST NOT** be defined in the same reference.  A Service Binding resource **MUST** define a `.spec.service` which is an `ObjectReference`-like declaration to a Provisioned Service-able resource.  A `ServiceBinding` **MAY** define the service reference by-name or by-[label selector][ls].  Extensions and implementations **MAY** allow additional kinds of applications and services to be referenced.  
+
+As label selectors are inherently queries that return zero-to-many resources, implementation **MUST** select the first match and ignore any other matches.  Therefore it is **RECOMMENDED** that `ServiceBinding` authors use a combination of labels that yield a single resource.  
```

Comment:
What about the zero matches scenario? I think the spec should add a mandate to yield at least one result for label selectors. So, the reconciler can raise error when there is no matching service found.